persist/fs/files.go

```diff
@@ -42,11 +42,105 @@ var timeZero time.Time
 
 const (
 	dataDirName       = "data"
+	snapshotDirName   = "snapshots"
 	commitLogsDirName = "commitlogs"
 )
 
 type fileOpener func(filePath string) (*os.File, error)
 
+// FilesetFile represents a set of Fileset files for a given block start
+type FilesetFile struct {
+	BlockStart time.Time
+	Files      []string
+}
+
+// FilesetFilesSlice is a slice of FilesetFile
+type FilesetFilesSlice []FilesetFile
+
+// Filepaths flattens a slice of FilesetFiles to a single slice of filepaths
+func (f FilesetFilesSlice) Filepaths() []string {
+	flattened := []string{}
+	for _, fileset := range f {
+		flattened = append(flattened, fileset.Files...)
+	}
+
+	return flattened
+}
+
+// NewFilesetFile creates a new Fileset file
+func NewFilesetFile(blockStart time.Time) FilesetFile {
+	return FilesetFile{
+		BlockStart: blockStart,
+		Files:      []string{},
+	}
+}
+
+// SnapshotFile represents a set of Snapshot files for a given block start
+type SnapshotFile struct {
+	// There can be multiple snapshot files with the same blockstart, so they
+	// also have an auto-incrementing index
+	Index int
+	FilesetFile
+}
+
+// HasCheckpointFile returns a bool indicating whether the given set of
+// snapshot files has a checkpoint file.
+func (s SnapshotFile) HasCheckpointFile() bool {
+	for _, fileName := range s.Files {
+		if strings.Contains(fileName, checkpointFileSuffix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// SnapshotFilesSlice is a slice of SnapshotFile
+type SnapshotFilesSlice []SnapshotFile
+
+// Flatten flattens a slice of SnapshotFiles to a single slice of filepaths
+func (f SnapshotFilesSlice) Flatten() []string {
```

Comment:
Hm any reason why this is called `Flatten(..)` vs `Filepaths(..)` which is what this method is named on the `FilesetFilesSlice` type?

Reply:
yeah prateek asked me to rename it, and I only renamed one of them. Will fix